src/main/java/org/damap/base/rest/dmp/service/DmpConsistencyUtility.java

```diff
@@ -147,6 +149,25 @@ private void enforceDatasetConsistency(DmpDO dmpDO) {
         datasetDO.setDateOfDeletion(null);
       }
     }
+
+    HashSet<String> seenDatasetDOI = new HashSet<>();
+    List<DatasetDO> uniqueDatasets = new ArrayList<>();
+
+    for (DatasetDO datasetDO : dmpDO.getDatasets()) {
+      if (
+              datasetDO.getSource() == EDataSource.REUSED &&
+              datasetDO.getDatasetId() != null &&
+              datasetDO.getDatasetId().getType() == EIdentifierType.DOI
+      ) {
+        String datasetDOI = datasetDO.getDatasetId().getIdentifier();
+        if (!seenDatasetDOI.contains(datasetDOI)) {
+            uniqueDatasets.add(datasetDO);
+            seenDatasetDOI.add(datasetDOI);
+        }
+      }
+    }
```

Comment:
major: This will also remove all new datasets instead of only filtering duplicates.

Reply:
Fixed it, was a missunderstanstind on my end and not completly thought through. Thanks for figuring it out!